rslib/src/storage/deck/mod.rs

```diff
@@ -210,6 +211,19 @@ impl SqliteStorage {
             .collect()
     }
 
+    pub(crate) fn immediate_child_decks(&self, parent: &Deck) -> Result<Vec<Deck>> {
+        let prefix_start = format!("{}\x1f", parent.name);
+        let prefix_end = format!("{}\x20", parent.name);
+        let child_descendant = format!("{}%\x1f%", escape_sql_wildcards(&prefix_start));
+        self.db
+            .prepare_cached(concat!(
+                include_str!("get_deck.sql"),
+                " where name >= ? and name < ? and not name like ? escape '\\'"
```

Comment:
Isn't the first point a show-stopper then?
What would be the alternative? Using `storage.child_decks()` and then figuring out the hierarchy in Rust? I guess this wouldn't be easy, but it would solve both points of concern.

Reply:
If you wanted to keep close to the current approach, I imagine you could emulate it by doing the not comparison in Rust after the rows are returned, or by using an extra sqlite function. But before going down that route, it may be worth considering whether we might be able to build our with a single query, similar to the way the deck list code does (IIRC it also deals with possible missing parents)

Reply:
So indeed using `storage.child_decks()` and then figuring out the hierarchy in Rust, only that you've already done the work which I can copy. 🙂
Wasn't easy to unify with the id_tree building logic, but it forced to to make some other simplifications.